common/namespace/registry.go

```diff
@@ -161,6 +167,11 @@ type (
 		cacheNameToID        cache.Cache
 		cacheByID            cache.Cache
 		stateChangeCallbacks map[any]StateChangeCallbackFn
+
+		// readthroughCache stores namespaces that missed the above caches
+		// AND was not found when reading through to the persistence layer
+		readthroughLock  sync.RWMutex
+		readthroughCache cache.Cache
```

Comment:
Using an LRU cache seems like an overkill to me. Could a map from id/name to lastReadThrough time work?
Or are you trying to handle million ns case?

Reply:
I think it's just for the free TTL